src/search.rs

```diff
@@ -0,0 +1,166 @@
+    async fn create_index_and_initialize_search(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<CreatedIndexResult>;
+}
+
+#[async_trait]
+impl BlindIndexSearchInitialize for IronOxide {
+    async fn initialize_search(
+        &self,
+        search: &EncryptedBlindIndexSalt,
+    ) -> Result<BlindIndexSearchSdk> {
+        let decrypted_value = self
+            .document_decrypt_unmanaged(
+                &search.encrypted_salt_bytes[..],
+                &search.encrypted_deks[..],
+            )
+            .await?;
+        decrypted_value.decrypted_data().try_into()
+    }
+    async fn create_index(&self, group_id: &GroupId) -> Result<EncryptedBlindIndexSalt> {
+        let CreatedIndexResult { encrypted_salt, .. } =
+            self.create_index_and_initialize_search(group_id).await?;
+        Ok(encrypted_salt)
+    }
+    async fn create_index_and_initialize_search(
+        &self,
+        group_id: &GroupId,
+    ) -> Result<CreatedIndexResult> {
+        let salt = {
+            let mut mut_salt = [0u8; 32];
+            take_lock(&self.rng).deref_mut().fill_bytes(&mut mut_salt);
+            mut_salt
+        };
+
+        let encrypted_salt = self
+            .document_encrypt_unmanaged(
+                &salt,
+                &DocumentEncryptOpts::with_explicit_grants(
+                    None,
+                    None,
+                    false,
+                    vec![group_id.into()],
+                ),
+            )
+            .await?;
+        let search = BlindIndexSearchSdk::new(salt);
+
+        Ok(CreatedIndexResult {
+            encrypted_salt: encrypted_salt.try_into()?,
+            sdk: search,
+        })
+    }
+}
+#[derive(Debug, PartialEq, Clone, Hash, Eq)]
+pub struct BlindIndexSearchSdk {
+    decrypted_salt: [u8; 32],
+}
+
+impl TryFrom<&[u8]> for BlindIndexSearchSdk {
+    type Error = IronOxideErr;
+    fn try_from(bytes: &[u8]) -> Result<BlindIndexSearchSdk> {
+        let decrypted_len = bytes.len();
+        if decrypted_len != REQUIRED_LEN {
+            std::result::Result::Err(IronOxideErr::WrongSizeError(
+                Some(decrypted_len),
+                Some(REQUIRED_LEN),
+            ))
+        } else {
+            let mut a = [0u8; 32];
+            a.copy_from_slice(&bytes[0..32]);
+            Ok(BlindIndexSearchSdk::new(a))
+        }
+    }
+}
+
+impl TryFrom<DocumentEncryptUnmanagedResult> for EncryptedBlindIndexSalt {
+    type Error = IronOxideErr;
+    fn try_from(r: DocumentEncryptUnmanagedResult) -> Result<EncryptedBlindIndexSalt> {
+        match r.access_errs().get(0) {
+            None => Ok(EncryptedBlindIndexSalt {
+                encrypted_deks: r.encrypted_deks().to_vec(),
+                encrypted_salt_bytes: r.encrypted_data().to_vec(),
+            }),
+            Some(err) => Err(IronOxideErr::UserOrGroupDoesNotExist(
+                err.user_or_group.clone(),
+            )),
+        }
+    }
+}
+
+impl BlindIndexSearchSdk {
```

Comment:
I think `BlindIndexSearch` is a slightly more succinct name with the same meaning

Reply:
ok.